openedx/core/djangoapps/schedules/config.py

```diff
@@ -9,4 +14,28 @@
     flag_undefined_default=False
 )
 
-DEBUG_MESSAGE_WAFFLE_FLAG = WaffleFlag(WAFFLE_FLAG_NAMESPACE, u'enable_debugging')
+DEBUG_NUDGE_MESSAGE_SAMPLE_NAME = u'schedules.enable_nudge_message_debugging'
+DEBUG_NUDGE = u'schedules.enable_nudge_debugging'
+
+
+@contextmanager
+def enable_debugging_if_necessary():
+    existing_level = None
+    if waffle.switch_is_active(DEBUG_NUDGE) or waffle.sample_is_active(DEBUG_NUDGE_MESSAGE_SAMPLE_NAME):
```

Comment:
One of the advantages to using our waffle_utils class is that it provides an extra caching layer - with the Request cache.  For some of our waffle usage, we were seeing a multitude of calls to Memcached.

Reply:
Unfortunately there is no request cache available in the mgmt command nor the celery worker as far as I know.

In fact, using the flag was causing problems since it was looking for `request.user` and wasn't able to find it. That's why I switched over to a switch and sample.

Now that I'm staring that this, though, I can probably just make it a single sample.